apps/files/lib/Provider/FileDownloadProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Louis Chmn <louis@chmn.me>
+ *
+ * @author Louis Chmn <louis@chmn.me>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\Files\Provider;
+
+use OCP\Files\Folder;
+use OCP\Files\Node;
+use OCP\Files\IFileDownloadProvider;
+
+class FileDownloadProvider implements IFileDownloadProvider {
+	private ?Folder $userFolder;
+
+	public function __construct(
+		?Folder $userFolder,
+	) {
+		$this->userFolder = $userFolder;
+	}
+
+	public function getNode(string $davPath): ?Node {
+		if (!str_starts_with($davPath, "files/")) {
+			return null;
+		}
+
+		if ($this->userFolder === null) {
+			return null;
+		}
+
+		/** @var ?string */
+		$userId = explode('/', $davPath)[1];
```

Comment:
Can save some work:
```suggestion
		$userId = explode('/', $davPath, 3)[1];
```

At the same time we can prevent a little log on invalid input:
```suggestion
		$userId = explode('/', $davPath, 3)[1] ?? null;
```